crates/primitives/src/transaction/error.rs

```diff
@@ -48,4 +48,8 @@ pub enum InvalidTransactionError {
     /// Thrown if the sender of a transaction is a contract.
     #[error("Transaction signer has bytecode set.")]
     SignerAccountHasBytecode,
+    /// Thrown if the transaction is a deposit transaction.
+    #[cfg(feature = "optimism")]
+    #[error("Deposit transactions cannot appear in the tx pool")]
+    PooledDepositTx,
```

Comment:
would `InvalidTransactionError::TxTypeNotSupported` be adequate here?

I will add another general purpose error variant here

Reply:
Yep, this sounds good. Let us know when it is in upstream and glad to swap over.

Reply:
@refcell I think `TxTypeNotSupported` covers the meaning of the error here - do we want a custom error here?